open_instruct/dpo_tune.py

```diff
@@ -744,6 +702,21 @@ def load_model():
             if len(tokenizer) > reference_embeddings.weight.shape[0]:
                 reference_model.resize_token_embeddings(len(tokenizer))
 
+    # set the tokenizer chat template to the training format
+    # this will be used for encoding the training examples
+    # and saved together with the tokenizer to be used later.
+    if args.chat_template_name in CHAT_TEMPLATES:
+        tokenizer.chat_template = CHAT_TEMPLATES[args.chat_template_name]
+    else:
+        try:
+            tokenizer.chat_template = AutoTokenizer.from_pretrained(args.chat_template_name).chat_template
+        except Exception as e:
+            raise ValueError(f"Could not find chat template for {args.chat_template_name}.")
+
+    if args.add_bos:
+        # also add bos in the chat template
+        tokenizer.chat_template = "{{ bos_token }}" + tokenizer.chat_template
```

Comment:
Should we assert `bos_token` not in `tokenizer.chat_template` if we decide to do this logic separately?